custom_components/zwave_mqtt/binary_sensor.py

```diff
@@ -52,27 +284,27 @@ def async_add_binary_sensor(values):
         sensors_to_add = []
 
         if values.primary.type == ValueType.LIST:
-
-            # Handle special cases
-            # we convert some of the Notification values into it's own binary sensor
+            # we convert Notification values into binary sensors
             # https://github.com/OpenZWave/open-zwave/blob/master/config/NotificationCCTypes.xml
-            # TODO: Use contants/Enums from lib (when added)
-            for item in values.primary.value["List"]:
-                if values.primary.index == 6 and item["Value"] == 22:
-                    # Door/Window Open
-                    sensors_to_add.append(
-                        ZWaveListValueSensor(values, item["Value"], DEVICE_CLASS_DOOR)
-                    )
-                if values.primary.index == 7 and item["Value"] in [7, 8]:
-                    # Motion detected
+            for list_value in values.primary.value["List"]:
+                # check if we have a mapping for this value
+                for item in NOTIFICATION_SENSORS:
+                    if item[NOTIFICATION_TYPE] != values.primary.index:
+                        continue
+                    if list_value["Value"] not in item[NOTIFICATION_VALUES]:
+                        continue
                     sensors_to_add.append(
-                        ZWaveListValueSensor(values, item["Value"], DEVICE_CLASS_MOTION)
+                        ZWaveListValueSensor(
+                            # required values
+                            values,
+                            list_value["Value"],
+                            # optional values
+                            item.get(NOTIFICATION_DEVICE_CLASS, None),
```

Comment:
`None` is the default value returned if the key is missing in the dict.
```suggestion
                            item.get(NOTIFICATION_DEVICE_CLASS),
```